### PR TITLE
Migrate diagrammatic equations

### PR DESCRIPTION
**Pull request by quffaro**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -10,6 +10,8 @@ Catlab = "134e5e36-593f-5add-ad60-77f754baafbe"
 CombinatorialSpaces = "b1c52339-7909-45ad-8b6a-6e388f7c67f2"
 ComponentArrays = "b0b7db55-cfe3-40fc-9ded-d10e2dbeff66"
 DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"
+DiagrammaticEquations = "6f00c28b-6bed-4403-80fa-30e0dc12f317"
```
Comment (lukem12345) [inline Project.toml:13]: There should be a [compat] entry for DiagrammaticEquations

File: examples/chemistry/brusselator_bounded.jl [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+  F = t -> t ≥ 1.1 ? F₂ : F₁)
+
+# Generate the simulation.
+gensim(expand_operators(Brusselator))
+sim = eval(gensim(expand_operators(Brusselator)))
+fₘ = sim(sd, generate)
+
+# Create problem and run sim for t ∈ [0,tₑ).
+# Map symbols to data.
+u₀ = construct(PhysicsState,
```
Comment (lukem12345) [inline examples/chemistry/brusselator_bounded.jl:138]: This should be updated to use the `ComponentArray` constructor as in the other examples.

File: src/Decapodes.jl [hunk 3]
```diff
@@ -7,40 +7,23 @@ using Catlab.CategoricalAlgebra
...
+export
+findname, flat_op,
+gensim, evalsim, closest_point, findnode, compile, compile_env,  PhysicsState,  default_dec_matrix_generate, default_dec_generate, default_dec_generate, VectorForm,
+AbstractMeshKey, loadmesh, Icosphere, Rectangle_30x10, Torus_30x10, Point_Map,
+CartesianPoint, SpherePoint, r, theta, phi, TangentBasis, θhat, ϕhat  
 
-normalize_unicode(s::String) = Unicode.normalize(s, compose=true, stable=true, chartransform=Unicode.julia_chartransform)
-normalize_unicode(s::Symbol)  = Symbol(normalize_unicode(String(s)))
-DerivOp = Symbol("∂ₜ")
 append_dot(s::Symbol) = Symbol(string(s)*'\U0307')
```
Comment (lukem12345) [inline src/Decapodes.jl:23]: Is this meant to stay here?

File: src/simulation.jl [hunk 4]
```diff
@@ -126,17 +126,17 @@ is_infer(d::SummationDecapode, var_name::Symbol) = is_infer(d, first(incident(d,
...
-    filter(parts(d, :Var)) do v
-        length(incident(d, v, :tgt)) == 0 &&
-        length(incident(d, v, :res)) == 0 &&
-        length(incident(d, v, :sum)) == 0 &&
-        d[v, :type] != :Literal
-    end
-end
-
-infer_state_names(d) = d[infer_states(d), :name]
+# TODO: Matt - Delete
```
Comment (lukem12345) [inline src/simulation.jl:129]: Delete

File: test/simulation.jl [hunk 5]
```diff
@@ -1,5 +1,6 @@
 using Decapodes
-import Decapodes: compile, gensim, infer_states, infer_state_names
+import Decapodes: compile, gensim
+# , infer_states, infer_state_names
```
Comment (lukem12345) [inline test/simulation.jl:3]: Delete this if not necessary.

**Comment by lukem12345** [comment]
You can fix the failing Docs build (aside from the random 404 error) by setting the `docs/Project.toml` compat entry for CairoMakie to 0.10

**Comment by jpfairbanks** [comment]
@lukem12345 @quffaro @olynch, can we make a full court press on this so that it doesn't start holding up changes to main?

File: test/composition.jl
Comment (lukem12345) [inline test/composition.jl:1]: @quffaro I don’t think this file is called by runtests in this branch’s current state. Is this meant to be in DgmEqns?

**Review by lukem12345 — CHANGES_REQUESTED** [review]

File: test/composition.jl
Comment (quffaro) [reply test/composition.jl:1]: Yup you're right. I've just re-tested with test/composition moved over to DiagrammaticEquations. I'll push onto DE#main and the re-commit to this PR.

Comment (lukem12345) [reply test/composition.jl:1]: Ok. If tests pass here and docs build correctly, then this is ready to approve upon the next review request

**Review by lukem12345 — APPROVED** [review]

**Pull request merged**